jbpm/jbpm-flow/src/test/java/org/jbpm/process/core/timer/BusinessCalendarImplTest.java

```diff
@@ -440,6 +440,102 @@ public void testCalculateSecondsBeforeEndHour() {
+        config.setProperty(BusinessCalendarImpl.END_HOUR, "18");
+        config.setProperty(BusinessCalendarImpl.HOLIDAYS, "2024-11-28:2024-11-29");
+
+        String currentDate = "2024-11-27 17:59:33.000";
+        String duration = "2H";
+        String expectedDate = "2024-12-02 15:59:33";
+
+        SessionPseudoClock clock = new StaticPseudoClock(parseToDateWithTimeAndMillis(currentDate).getTime());
+        BusinessCalendarImpl businessCal = new BusinessCalendarImpl(config, clock);
+
+        Date result = businessCal.calculateBusinessTimeAsDate(duration);
+
+        assertThat(formatDate("yyyy-MM-dd HH:mm:ss", result)).isEqualTo(expectedDate);
+    }
+
+    @Test
+    public void testCalculateHoursToExecuteSameDayStartingBeforeWorkingOurs() {
+        Properties config = new Properties();
+        config.setProperty(BusinessCalendarImpl.HOURS_PER_DAY, "4");
+        config.setProperty(BusinessCalendarImpl.START_HOUR, "14");
+        config.setProperty(BusinessCalendarImpl.END_HOUR, "18");
+
+        String currentDate = "2024-11-27 13:59:33.000";
+        String duration = "2H";
+        String expectedDate = "2024-11-27 16:59:33";
+
+        SessionPseudoClock clock = new StaticPseudoClock(parseToDateWithTimeAndMillis(currentDate).getTime());
+        BusinessCalendarImpl businessCal = new BusinessCalendarImpl(config, clock);
+
+        Date result = businessCal.calculateBusinessTimeAsDate(duration);
+
+        assertThat(formatDate("yyyy-MM-dd HH:mm:ss", result)).isEqualTo(expectedDate);
+    }
+
+    @Test
+    public void testCalculateHoursToExecuteNextDayStartingBeforeWorkingOurs() {
+        Properties config = new Properties();
+        config.setProperty(BusinessCalendarImpl.HOURS_PER_DAY, "4");
+        config.setProperty(BusinessCalendarImpl.START_HOUR, "14");
+        config.setProperty(BusinessCalendarImpl.END_HOUR, "18");
+
+        String currentDate = "2024-11-27 13:59:33.000";
+        String duration = "5H";
+        String expectedDate = "2024-11-28 14:59:33";
+
+        SessionPseudoClock clock = new StaticPseudoClock(parseToDateWithTimeAndMillis(currentDate).getTime());
+        BusinessCalendarImpl businessCal = new BusinessCalendarImpl(config, clock);
+
+        Date result = businessCal.calculateBusinessTimeAsDate(duration);
+
+        assertThat(formatDate("yyyy-MM-dd HH:mm:ss", result)).isEqualTo(expectedDate);
+    }
+
+    @Test
+    public void testCalculateHoursToExecuteAfterHolidaysStartingBeforeWorkingOurs() {
+        Properties config = new Properties();
+        config.setProperty(BusinessCalendarImpl.HOURS_PER_DAY, "4");
+        config.setProperty(BusinessCalendarImpl.START_HOUR, "14");
+        config.setProperty(BusinessCalendarImpl.END_HOUR, "18");
+        config.setProperty(BusinessCalendarImpl.HOLIDAYS, "2024-11-28:2024-11-29");
+
+        String currentDate = "2024-11-27 13:59:33.000";
+        String duration = "5H";
+        String expectedDate = "2024-12-02 14:59:33";
```

Comment:
```
expected: "2024-12-02 14:59:33"
 but was: "2024-12-02 15:00:00"
```
The actual returned result `2024-12-02 15:00:00` is correct here. The logic is:
* The calculation starts outside working hours
* Starting point is shifted to the next working hours `2024-11-27 14:00:00`
* On that day, 4h of the 5h timer are within working hours, which leaves exactly 1h still to be scheduled
* 11-28 and 11-29 are holidays
* 11-30 and 12-01 are weekend days
* 12-02 is the next working day starting at 14:00
* Expected date is 2024-12-02 14:00:00 plus the 1hr that hasn't been scheduled during working hours